### ISSUE TITLE
CLI support non-existent .eslintrc

### ISSUE THREAD

**Issue by reviewher** [eslint/eslint#11026]
**The version of ESLint you are using.** 5.3.0

**The problem you want to solve.** Run ESLint on a project that doesn't currently have a `.eslintrc`

**Your take on the correct solution to problem.**

ESLint currently errors when running on a project that has no `.eslintrc` file where none of the parent directories have a `.eslintrc`.  However, ESLint supports an empty `.eslintrc` file (no content) that you can produce with `touch .eslintrc`.

Given that there's a clear "default behavior", instead of erroring, when the file doesn't exist ESLint should warn users that the `.eslintrc` file does not exist but behave as if an empty `.eslintrc` file was available

**Are you willing to submit a pull request to implement this change?** Yes

**Comment by reviewher** [comment]
@kaicataldo I agree that the important thing is for both behaviors to be consistent (either error on empty file OR support non-existent file)

There may not be a default ruleset, but there is definitely a default parser configuration.  For example, https://github.com/kirillmurashov/vue-drag-resize/ has no .eslintrc.  Running will give the standard missing configuration error.  Adding an empty .eslintrc or empty object (`{}`) shows parsing errors pertaining to `export` and `import`.  Whether this is useful is up for debate, but it seems that this is more than merely a "noop"

**Comment by kaicataldo** [comment]
@reviewher Understood - I didn't realize that was the current behavior. I could see the argument being made that someone might want to just validate that they have valid syntax, however I think using Espree or Acorn directly would be a better way to do that. I still think we should probably throw an error for empty config files, but let's see what the rest of the team thinks.

**Comment by not-an-aardvark** [comment]
I don't feel very strongly about it, but I think the current behavior of only doing syntax validation for an empty config is logically consistent and sometimes useful. Given that all rules are off by default, it makes sense to me that if you create a config without any rules, then ESLint will run as before with zero rules enabled.

In the past, I've added ESLint to a project to just to have syntax checking at first -- it's true that someone could use Espree or Acorn directly, but ESLint has some useful CLI features that aren't supported by either of those projects. (Actually, when I did this I used `--no-eslintrc` rather than providing an empty config file, although I'm not sure having two ways to do something is necessarily a problem in this case.)

Do we have reason to believe that the behavior of empty config files is causing confusion for users? My preference would be to leave it as-is if it's not causing confusion, because throwing an error seems like it would be adding a special case.

**Comment by reviewher** [comment]
ESLint clearly has a "default configuration" for empty config, which means that there is a meaningful behavior for a missing config.  As a result, the default for a project without a `.eslintrc` probably should be a warning (with body text similar to the current error behavior) followed by execution as if the config was empty.  The `--no-eslintrc` flag would suppress the warning message.

**Comment by kaicataldo** [comment]
One more thing to consider here is that not all empty config files are treated equally. An `.eslintrc.json` filet that contains no content throws the error 
```
Cannot read config file: /path/to-file/.eslintrc.json
Error: Unexpected end of JSON input
``` 
while `.eslintrc` does not. I believe this is due to the latter two being valid syntax for an empty YAML file while the former is not valid JSON. Adding `{}` to the empty `.eslintrc.json` file makes the error go away.

This makes things trickier since we support more than one file type for our configs, and I'm starting to think that the current behavior is actually the right one (so that we don't make any assumptions). Whatever we decide here, I just want to make sure the current behavior is by design and not by accident!

**Comment by nzakas** [comment]
Another thing to consider: `.eslintrc` is a deprecated feature in ESLint. We don't allow people to create it using `--init` and would prefer that people use the formal JSON, YAML, or JS versions of that file instead.

The behavior noted in this issue was originally by design, when we implemented the requirement that a config must be present to run ESLint. We intentionally said that the presence of a config file, even if empty, means the user wants to run ESLint with default settings.

Given that, I think the only real question here is if parsing an empty `.eslintrc` should have the same result as parsing an empty `.eslintrc.json`. I can see the argument that it should, though I don't feel strongly about that.
